src/open_company_web/components/topics_columns.cljs

```diff
@@ -112,26 +114,48 @@
       (= min-height thrd-clmn)
       :3)))
 
+(defn get-pinned-layout [pinned-topics columns-num]
+  (if (= columns-num 3)
+    (loop [idx 3
+           cl1 (vec (remove nil? [(first pinned-topics)]))
+           cl2 (vec (remove nil? [(second pinned-topics)]))
+           cl3 (vec (remove nil? [(get pinned-topics 2)]))]
+      (if (<= idx (count pinned-topics))
+        (recur (+ idx 3)
+               (vec (remove nil? (conj cl1 (get pinned-topics idx))))
+               (vec (remove nil? (conj cl2 (get pinned-topics (inc idx)))))
+               (vec (remove nil? (conj cl3 (get pinned-topics (+ idx 2))))))
+        {:1 cl1
+         :2 cl2
+         :3 cl3}))
+    (loop [idx 2
+           cl1 (vec (remove nil? [(first pinned-topics)]))
+           cl2 (vec (remove nil? [(second pinned-topics)]))]
+      (if (<= idx (count pinned-topics))
+        (recur (+ idx 2)
+               (vec (remove nil? (conj cl1 (get pinned-topics idx))))
+               (vec (remove nil? (conj cl2 (get pinned-topics (inc idx))))))
+        {:1 cl1
+         :2 cl2}))))
+
 (defn calc-layout [owner data]
```

Comment:
function docstring and possibly some inline comments would be good here. this function is non-trivial.